ops/main.py

```diff
@@ -144,6 +144,10 @@ def _emit_charm_event(charm: 'ops.charm.CharmBase', event_name: str):
         event_to_emit.emit(*args, **kwargs)
 
 
+def _get_juju_relation_id():
```

Comment:
Splitting it out was actually a code review suggestion from @benhoyt :smile:. I'm weakly in favour of splitting it out for the reasons you mention.

Note that we have tentative plans to refactor the env parsing (#1075) and if that goes ahead then this would presumably end up in the initialisation of that context object and no longer need this little helper.